components/Predict.tsx

```diff
@@ -390,193 +375,23 @@ export function Predict({
-                    )}
-                  >
-                    %
-                  </span>
-                </div>
-              </div>
-
-              {embedded && (
-                <div className="flex items-center">
-                  <label
-                    htmlFor="sharePublicly"
-                    className="text-sm max-w-[8rem] ml-2"
-                  >
-                    Share with anyone with the link?
-                  </label>
-                  <input
-                    type="checkbox"
-                    id="sharePublicly"
-                    defaultChecked={
-                      typeof window !== "undefined" &&
-                      window.localStorage.getItem("lastSharedPubliclyState") ===
-                        "true"
-                    }
-                    className="ml-2 checkbox check"
-                    onClick={(e) => {
-                      // NB: only works per-site, but better than nothing
-                      localStorage.setItem(
-                        "lastSharedPubliclyState",
-                        e.currentTarget.checked ? "true" : "false",
-                      )
-                    }}
-                    onMouseDown={(e) => e.stopPropagation()}
-                    {...register("sharePublicly")}
-                  />
-                </div>
-              )}
-            </div>
-
-            <div className="self-center">
-              <button
-                onClick={(e) => {
-                  e.preventDefault()
-                  void handleSubmit(onSubmit, () => {
-                    // on invalid:
-                    if (!userId) {
-                      if (embedded) {
-                        window.open("https://fatebook.io", "_blank")?.focus()
-                      } else {
-                        void signInToFatebook()
-                      }
-                    }
-                  })()
-                }}
-                className="btn btn-primary btn-lg hover:scale-105"
-                disabled={
-                  !!userId && Object.values(errors).some((err) => !!err)
-                }
-              >
-                {userId || session.status === "loading"
-                  ? "Predict"
-                  : "Sign up to predict"}
-              </button>
-            </div>
-          </div>
+          <BinaryQuestion
+            small={small}
+            resolveByButtons={resolveByButtons}
+            questionDefaults={questionDefaults}
+            embedded={embedded}
+            userId={userId}
+            onSubmit={onSubmit}
+            signInToFatebook={signInToFatebook}
+            session={session}
+            register={register}
+            setValue={setValue}
+            errors={errors}
+            watch={watch}
+            handleSubmit={handleSubmit}
+            textAreaRef={textAreaRef}
+            highlightResolveBy={highlightResolveBy}
+          ></BinaryQuestion>
```

Comment:
nit: I generally prefer `<Component prop={x} />` for components with no children for conciseness